pkg/helm/chartproxy/proxy.go

```diff
@@ -91,13 +91,23 @@ func (p *proxy) IndexFile(onlyCompatible bool, namespace string) (*repo.IndexFil
 
 			for key, entry := range idxFile.Entries {
 				for i := len(entry) - 1; i >= 0; i-- {
```

Comment:
This can be rewritten as:

```go
for _, version := range entry { // entry could be renamed to `versions`
    if version.Annotations == nil {
        version.Annotations = map[string]string{}
    }
    version.Annotations["repositoryName"] = helmRepo.Name
    version.Annotations["repositoryNamespace"] = helmRepo.Namespace

    ...
}
```